src/internal/clusterstate/v2.11.0/commit_totals.go

```diff
@@ -0,0 +1,54 @@
+package v2_10_0
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/migrations"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"go.uber.org/zap"
+	"strings"
+)
+
+func normalizeCommitTotals(ctx context.Context, env migrations.Env) error {
+	ctx = pctx.Child(ctx, "normalizeCommitTotals")
+	tx := env.Tx
+	if _, err := tx.ExecContext(ctx, `ALTER TABLE pfs.commits ADD COLUMN total_fileset_id UUID REFERENCES storage.filesets(id)`); err != nil {
+		return errors.Wrap(err, "add total_fileset column to pfs.commits")
+	}
+	count := int64(0)
+	if err := tx.GetContext(ctx, &count, `SELECT count(fileset_id) FROM pfs.commit_totals;`); err != nil {
+		return errors.Wrap(err, "counting rows in pfs.commit_totals")
+	}
+	if count == 0 {
+		return nil
+	}
+	log.Info(ctx, "normalizing pfs.commit_totals", zap.Int64("total", count))
+	// If there are commits in the pfs.commit_totals table that do not exist in the pfs.commits table, bail on the migration.
```

Comment:
Hmm, I think we may be better off just dropping them in this case since they aren't accessible without corresponding entries in the commits table. Is there a reason to believe that this may be an issue?

Reply:
I don't think this will lead to an issue since we want the pfs.commits to be the source of truth. I just wanted to be conservative with the migration.

Reply:
Wouldn't the outcome just be that the migration would fail without any path to recovery? I think we need to just ignore the dangling commits in that case, because there is nothing we can do if that happens.